packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -366,7 +369,22 @@ class DatabaseConnection {
     Transaction? transaction,
     bool ignoreRows = false,
     bool simpleQueryMode = false,
+    Object? parameters,
   }) async {
+    assert(
+      parameters == null ||
+          parameters is List<Object?> ||
+          parameters is Map<String, Object?> ||
+          parameters is QueryParameters,
+      'Parameter type should be one of: QueryParameters || List<Object?> || Map<String, Object?>',
+    );
+
+    assert(
+      simpleQueryMode == false ||
+          (simpleQueryMode == true && parameters == null),
+      'simpleQueryMode does not support parameters',
+    );
```

Comment:
Suggestion: If this is needed, please change this to a throw so that we don't allow this in production systems either.

This should also have a test so we validate that it is enforced if someone refactors the class later on.

Reply:
The Postgres driver will throw an error on this condition. This is a bit deeper down the code, but it should do the job. 

Doing extra checks for production adds up every single time by a tiny bit. Do you still want to do this?

Reply:
Actually, I could just add a unit test for this without adding an extra run time check of our own and remove the assertion. I suppose that would make both of us happy 😀

Reply:
Turns out to be rather difficult to test this, since `DatabaseConnection` is hidden behind the Database object.  We're not passing arguments in `simpleQuery` either, so there is no way to force an exception from that level. Any ideas?

Reply:
Actually, it might be OK to have an assert here. Since we don't expose the possibility of making the wrong decision (the result of this being hard to test), the check is currently only for us internally when we develop new interfaces.

You can leave this as is :)